File: ClientApp/src/app/Types/Estoque.ts
import { Saida } from './Saida';
import { Entrada } from './Entrada';
import { Usuario } from './Usuario';
import { Alimento } from './Alimento';

export interface Estoque {
    Id: number;
    IdEntrada?: string;
    IdSaida?: string;
    IdAutor: number;
    Data: Date;
    IdAlimento: number;
    Quantidade: number;

    Saida: Saida;
    Entrada: Entrada;
    Autor: Usuario;
    Alimento: Alimento;
}

File: Model/Estoque.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AjudaMusica.Model
{
    public class Estoque
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public Guid? IdEntrada { get; set; }
        public Guid? IdSaida { get; set; }
        public int IdAutor { get; set; }
        public DateTime Data { get; set; }
        public int IdAlimento { get; set; }
        public int Quantidade { get; set; }

        [ForeignKey("IdEntrada")]
        public vwEntrada Entrada {get;set;}
        
        [ForeignKey("IdSaida")]
        public Saida Saida {get;set;}

        [ForeignKey("IdAutor")]
        public Usuario Autor {get;set;}
        
        [ForeignKey("IdAlimento")]
        public Alimento Alimento {get;set;}
    }
}
